### PR TITLE
Fixed crashes when the user opens form with unexpected element in 'field-list'

### LINKED ISSUE #2828: ODK Collect crashes when the user opens form with unexpected element in 'field-list'
#### Software and hardware versions 
Collect v1.18.0, v1.19
Error is visible on v1.17

#### Problem description
ODK Collect crashes when the user opens form with unexpected element in 'field-list'

#### Steps to reproduce the problem
1. User opens form 
[g6Error.xml.txt](https://github.com/opendatakit/collect/files/2795888/g6Error.xml.txt)
2. App crashes

#### Expected behavior
Error appears
06e67

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #2828

#### What has been done to verify that this works as intended?
I tested forms attached below:

#### Why is this the best possible solution? Were any other approaches considered?
The problem was that the previous approach we used to determine if a group is empty:
`getQuestionPrompts().length > 0`
throws an exception if a group contains not allowed elements but we don't care about it now we want to know if it contains any elements no matter they are valid or not.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It should just fix the problem. It shouldn't affect anything else.
[emptyGroupFieldList.xml.txt](https://github.com/opendatakit/collect/files/2796370/emptyGroupFieldList.xml.txt)
[emptyGroupFieldList2.xml.txt](https://github.com/opendatakit/collect/files/2796371/emptyGroupFieldList2.xml.txt)
[g6Error.xml.txt](https://github.com/opendatakit/collect/files/2796372/g6Error.xml.txt)
[g6Error2.xml.txt](https://github.com/opendatakit/collect/files/2796373/g6Error2.xml.txt)

@mmarciniak90 please play with different forms a little bit because I noticed some strange behavior for example in form g6Error2.xml not related to this issue and it doesn't cause any crash but still it's strange and I think we need to add some improvements.
Eg. We need to swipe twice to skip a group that contains not allowed elements if that group is not at the beginning of a form. Maybe you can find something else.

#### Do we need any specific form for testing your changes? If so, please attach one.
I tested:

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/logic/FormController.java [hunk 1]
```diff
@@ -833,6 +833,10 @@ public void setLanguage(String language) {
         return questions;
     }
 
+    private boolean isGroupEmpty() {
+        return getIndicesForGroup((GroupDef) formEntryController.getModel().getForm().getChild(getFormIndex())).isEmpty();
```
Comment (cooperka) [inline collect_app/src/main/java/org/odk/collect/android/logic/FormController.java]: This line is a bit much, mind factoring out<br/>`GroupDef group = (GroupDef) formEntryController.getModel().getForm().getChild(getFormIndex());` ?

**Review by cooperka — COMMENTED** [review]
LGTM besides nit. I also see `getQuestionPrompts().length != 0` used in `stepToNextScreenEvent()`, shall we update that as well?

File: collect_app/src/main/java/org/odk/collect/android/logic/FormController.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/logic/FormController.java]: Improved.

**Comment by grzesiek2010** [comment]
> I also see getQuestionPrompts().length != 0 used in stepToNextScreenEvent(), shall we update that as well?

Maybe, I'm not sure `stepToNextScreenEvent() throws JavaRosaException` so it might be enough but I asked @mmarciniak90 to test some strange forms.

**Review by cooperka — APPROVED** [review]

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android: 4.2, 4.4, 5.1, 6.0, 7.0, 8.1
Verified cases:
- verified 4 forms added by Grzegorz
- noticed strange behavior mentioned by Grzegorz but this case works better with fix than without it
- verified other forms and I did not notice regression
- new and saved form
- rotating, locking

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Review by yanokwa — APPROVED** [review]

**Comment by yanokwa** [comment]
Thanks @mmarciniak90! @grzesiek2010, please file an issue about the strange behavior you noticed so we have it on record.

**Pull request merged**